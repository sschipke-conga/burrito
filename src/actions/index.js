export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const setOrder = order => ({
  type: 'SET_ORDER',
  order
})
