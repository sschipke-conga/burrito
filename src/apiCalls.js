export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}

export const postOrder = async (order) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(order),
    headers: {
      'Content-Type': 'application/json'
    }
  };

  let res = await fetch('http://localhost:3001/api/v1/orders', options)
  return res.json()
}

export const removeOrder = async id => {
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  fetch(`http://localhost:3001/api/v1/orders/${id}`, options)
}