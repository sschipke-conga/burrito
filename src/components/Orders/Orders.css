.order {
  border: 1px solid black;
  display: inline-block;
  margin: 10px;
  background-color: #FFF;
}

.ingredient-list {
  list-style-type: none;
  padding: 0;
  width: 150px;
}

.no-ingredients {
  color: red;
}