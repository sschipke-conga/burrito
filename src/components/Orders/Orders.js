import React, {Component} from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions';
import { getOrders } from '../../apiCalls';

import './Orders.css';


export class Orders extends Component { 

  componentDidMount() {
    getOrders()
      .then(data => this.props.setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

    render() {
      const orderEls = this.props.orders.map((order, i) => {
        return (
          <div key={i} className="order">
            <h3>{order.name}</h3>
            <ul className="ingredient-list">
              {order.ingredients.map((ingredient, i) => {
                return <li key={i}>{ingredient}</li>
              })}
            </ul>
          </div>
        )
      });
    return (
      <section>
        { orderEls.length ? orderEls : <p>No orders yet!</p> }
      </section>
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);