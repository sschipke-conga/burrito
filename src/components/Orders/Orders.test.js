import React from 'react';
import { Orders, mapDispatchToProps, mapStateToProps } from './Orders';
import { shallow } from 'enzyme';
import { setOrders } from '../../actions';
import { getOrders } from '../../apiCalls';

jest.mock('../../apiCalls')
const mockOrders = [{
  name: 'Alex', ingredients: ['queso', 'pico', 'cheese']
}, {
  name: 'Steve', ingredients: ['garlic', 'bread', 'bagels']
}]

const mockSetOrders = jest.fn();
getOrders.mockImplementation(() =>{ 
  return Promise.resolve({orders: Promise.resolve(mockOrders)}) 
})

describe('Orders', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Orders 
      orders={mockOrders}
    setOrders={mockSetOrders}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should call getOrders when mounted', () => {
    expect(getOrders).toHaveBeenCalled()
  })
  describe('mapDispatchToProps and mapStateToProps', () => {
    it('should call Dispatch with setQuestions actions when setQuestions is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setOrders('SET_ORDERS', mockOrders)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setOrders('SET_ORDERS', mockOrders)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
    it('mapStateToProps gives all orders', () => {
      const mockState = {
        orders:mockOrders
      };
      const expected = {
        orders: mockState.orders
      };
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})