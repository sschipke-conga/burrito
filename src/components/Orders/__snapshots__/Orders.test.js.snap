// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="0"
  >
    <h3>
      Alex
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        queso
      </li>
      <li
        key="1"
      >
        pico
      </li>
      <li
        key="2"
      >
        cheese
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="1"
  >
    <h3>
      Steve
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        garlic
      </li>
      <li
        key="1"
      >
        bread
      </li>
      <li
        key="2"
      >
        bagels
      </li>
    </ul>
  </div>
</section>
`;
